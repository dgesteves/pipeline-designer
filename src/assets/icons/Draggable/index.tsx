import { TProps } from "../types";

export default function Draggable(props: TProps): JSX.Element {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="1em"
      height="1em"
      viewBox="0 0 32 32"
      xmlSpace="preserve"
      {...props}
    >
      <path
        style={{
          fill: "currentColor",
        }}
        d="M10 6h4v4h-4zM18 6h4v4h-4zM10 14h4v4h-4zM18 14h4v4h-4zM10 22h4v4h-4zM18 22h4v4h-4z"
      />
      <path
        style={{
          fill: "none",
        }}
        d="M0 0h32v32H0z"
      />
    </svg>
  );
}
