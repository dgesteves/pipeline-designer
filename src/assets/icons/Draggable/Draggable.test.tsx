import { render } from "@testing-library/react";
import Draggable from "./index";

describe("Draggable icon", () => {
  it("should render correctly", () => {
    const { container } = render(<Draggable />);
    expect(container).toBeInTheDocument();
  });

  it("should have the correct svg", () => {
    const { container } = render(<Draggable />);
    expect(container).toMatchSnapshot();
  });
});
