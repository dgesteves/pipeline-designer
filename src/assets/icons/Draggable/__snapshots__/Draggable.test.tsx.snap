// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draggable icon should have the correct svg 1`] = `
<div>
  <svg
    height="1em"
    viewBox="0 0 32 32"
    width="1em"
    xml:space="preserve"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 6h4v4h-4zM18 6h4v4h-4zM10 14h4v4h-4zM18 14h4v4h-4zM10 22h4v4h-4zM18 22h4v4h-4z"
      style="fill: currentColor;"
    />
    <path
      d="M0 0h32v32H0z"
      style="fill: none;"
    />
  </svg>
</div>
`;
